server/auth/store.go

```diff
@@ -216,14 +218,18 @@ type AuthBatchTx interface {
 }
 
 type AuthReadTx interface {
+	RLock()
+	RUnlock()
+	UnsafeRead
+}
+
+type UnsafeRead interface {
```

Comment:
Suggestion: rename to `UnsafeAuthReadTx` or `UnsafeReadTx`, because `UnsafeRead` looks like a method name

Reply:
Name intentionally skipped `Tx` in name.`UnsafeRead` is a group of methods for reading in unsafe way. I wanted to differentiate fact that is no longer a transaction, because it doesn't give access to `Lock`/`Unlock`. This means that consumer doesn't control other changes that can go the transaction, aka not a transaction.

Reply:
> UnsafeRead is a group of methods for reading

Interface provide a way to specify the behavior of an object. It should be named in a declarative way (e.g. `UnsafeReadTx`), instead of imperative (e.g. `UnsafeRead`).

> because it doesn't give access to Lock/Unlock

It's why it's `Unsafe*Tx`

Reply:
> It should be named in a declarative way (e.g. UnsafeReadTx), instead of imperative (e.g. UnsafeRead).

Don't think this has any relation to being declarative vs imperative. Both words are noun. "Read" is a noun, "Txn" is a noun. When you combine them "read txn", then "read becomes a adjective while "txn" remains a noun. Overall it's a correct english. I explicitly want to under underscore that read functions under this interface are just reads and not transactions. 

> It's why it's Unsafe*Tx

No, "unsafe" means that consumer needs to call `Lock` before.  You are mixing consumer of "UnsafeRead" interface, which is a function, and the caller of the `UnsafeX` function which can takes "UnsafeRead" as argument.

Reply:
Read https://go.dev/doc/effective_go#interface-names

Reply:
Ok, that's separate independent convention for Go. To use `agent noun` instead of just `noun`. I can update `UnsafeRead` to `UnsafeReader`.

Reply:
Right, `UnsafeReader` is much better.